app/validation/validator.py

```
@@ -122,80 +122,79 @@ def validate_answer(self, block, answer):
-            return [self._error_message(error_message)]
-
-        error_message = 'Value {} used in {} should be between system limits {} to {}'\
-            .format(value, answer_id, MIN_NUMBER, MAX_NUMBER)
-
-        if MIN_NUMBER <= value <= MAX_NUMBER:
-            return []
-
-        return [self._error_message(error_message)]
+    def _get_answer_range_values(self, answer, answer_ranges):
+
+        return {
+            'min': self._get_answer_minimum_or_maximum(answer['id'],
+                                                       answer_ranges,
+                                                       answer.get('min_value', {'value': 0}),
+                                                       answer.get('decimal_places', 0),
+                                                       'minimum'),
+            'max': self._get_answer_minimum_or_maximum(answer['id'],
+                                                       answer_ranges,
+                                                       answer.get('max_value', {'value': MAX_NUMBER}),
+                                                       answer.get('decimal_places', 0),
+                                                       'maximum'),
+            'decimal_places': answer.get('decimal_places', 0),
+        }
+
+    def _get_answer_minimum_or_maximum(self, answer_id, answer_ranges, attributes, decimal_places, min_or_max):
+        values = []
+        if 'value' in attributes:
+            if not MIN_NUMBER <= attributes.get('value') <= MAX_NUMBER:
+                error_message = 'Value {} used in {} should be between system limits {} to {}' \
+                    .format(attributes['value'], answer_id, MIN_NUMBER, MAX_NUMBER)
+                raise ValidationError(error_message)
+            values = [attributes.get('value')]
+
+        elif 'answer_id' in attributes:
+            if decimal_places < answer_ranges[attributes['answer_id']]['decimal_places']:
```

what if the `answer_id` isn't in `answer_ranges`

---

if the answer id is not an existing id of an answer, the method `_get_rsnge_for_referred_answer` will raise an erro

---

which is called after this?

---

Yes but this checks within a `min_value` or a `max_value`, so in that case it will always be referring a previous answer_id

---

your assuming its always a previous answer. if the schema was incorrect ( which is likely hence we are validating it) then we would want to return an error saying that you are referencing  future answer rather then blowing up with an index error

---

Right so it should go off to validate that answer_id exists before the decimal places validation

---

would it work if you moved `_get_range_for_referred_answer` before this?